client/extensions/woocommerce/woocommerce-services/state/shipping-label/reducer.js

```diff
@@ -1346,8 +1346,14 @@ reducers[ WOOCOMMERCE_SERVICES_SHIPPING_LABEL_CLOSE_DETAILS_DIALOG ] = state =>
 };
 
 // Reset the state when the order changes
-reducers[ WOOCOMMERCE_ORDER_UPDATE_SUCCESS ] = () => {
-	return initializeLabelsState();
+reducers[ WOOCOMMERCE_ORDER_UPDATE_SUCCESS ] = ( { fulfillOrder }, { order: { status } } ) => {
+	const initialState = initializeLabelsState();
+
+	if ( fulfillOrder && "completed" === status ) {
+		initialState[ 'shouldUpdateOrderDetailPage' ] = true;
+	}
+
+	return initialState;
```

Comment:
If we marked the option "mark order as complete" and upon order update success, we got the order state as "completed", we should update the order's detail page.

Not sure here if the state associated to the shipping label should hold this variable or if it should be move to another section of the state, though.